### PR TITLE
Fix san issues in cluster

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR: https://github.com/arangodb/enterprise/pull/1186
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Sharding/ShardingStrategyDefault.h [hunk 1]
```diff
@@ -99,13 +99,15 @@ class ShardingStrategyHashBase : public ShardingStrategy {
...
 
   ShardingInfo* _sharding;
-  std::vector<ShardID> _shards;
   bool _usesDefaultShardKeys;
-  std::atomic<bool> _shardsSet;
-  Mutex _shardsSetMutex;
+
+ private:
+  std::atomic_bool _shardsSet;
+  std::mutex _shardsSetMutex;
```
Comment (jsteemann) [inline arangod/Sharding/ShardingStrategyDefault.h:109]: Should this header now include `<mutex>` and stop including `Basics/Mutex.h`?

Comment (MBkkt) [reply arangod/Sharding/ShardingStrategyDefault.h:109]: Yep

Comment (MBkkt) [reply arangod/Sharding/ShardingStrategyDefault.h:109]: Fixed

**Review by jsteemann — APPROVED** [review]
LGTM

**Review by gnusi — APPROVED** [review]

**Comment by MBkkt** [comment]
https://github.com/arangodb/arangodb/pull/17989

**Pull request merged**